### ISSUE TITLE
Add "Retrieving new messages" status to sync area

### ISSUE THREAD

**Issue by eloquence** [freedomofpress/securedrop-client#775]
As originally discussed [here](https://github.com/freedomofpress/securedrop-ux/issues/97#issuecomment-576891785), we'd ideally like to give the user some indication during a sync if new information is being downloaded from the server. This is especially important during client startup and for busy servers, so the user can see as quickly as possible whether or not they should shut down the client or wait for new content to arrive.

Other than the "Last Refresh:" date when there's no connectivity (#747), this is the _only_ purpose for which we want to use the area next to the network activity icon (i.e. the status text updated via `update_activity_status`).

We may want to tweak the exact wording and behavior a bit, but for now we can go with "Retrieving new messages" as the status message. While the retrieval is ongoing, the activity indicator should be animated using [this](https://github.com/freedomofpress/securedrop-ux/issues/17#issuecomment-578330193) GIF, continuously looping until the sync is complete. 

To ensure the user can at least spot the text and the animation doesn't flicker, we should set a minimum duration for text and animation. We can play with the exact value; I would suggest using around 250ms as the _minimum_ duration for now (there is no maximum).

tl;dr, to resolve this issue:
- "Retrieving new messages" should show while we're getting new messages or replies, not before or after
- Animation should loop continuously while this is happening (not flicker on and off during retrieval, nor flicker too quickly)
- We will want to run the initial one-time animation for the sync itself

Out of scope for now:
- Counting up messages or replies (a basic count like "(1/232)" would be a nice follow-up iteration)
- Showing any "Hooray! We've got new stuff" message at the end

**Comment by creviera** [comment]
just a headsup for this issue, we currently show "Downloading new messages" if there are any new messages that we are about to download, which we talked about removing here: https://github.com/freedomofpress/securedrop-client/blob/fb745f218935559d2e703a4303f429bc1750dd81/securedrop_client/logic.py#L548-L549

**Comment by creviera** [comment]
We've had engineering discussions in the past about how we would want to implement this. Here's the proposal:

1. If we see that there are new messages to download (https://github.com/freedomofpress/securedrop-client/blob/fb745f218935559d2e703a4303f429bc1750dd81/securedrop_client/logic.py#L548-L549), we set the activity status to "Retrieving new messages" and send a signal to the GUI to start animating the sync icon.

2. In this loop: https://github.com/freedomofpress/securedrop-client/blob/fb745f218935559d2e703a4303f429bc1750dd81/securedrop_client/logic.py#L551-L552, we mark the last job as the `last_job_of_a_sync` and send a signal from the job and then from the Controller that tells the GUI to stop the animation.

For step one, we'll want to ensure that this "Retrieving new messages" activity status remains until the last sync job finishes and the animation stops. so we will want to update the ActivityStatusBar to be similar to the ErrorStatusBar in that passing a zero duration means we display the message until it's set to a new message: https://github.com/freedomofpress/securedrop-client/blob/fb745f218935559d2e703a4303f429bc1750dd81/securedrop_client/gui/widgets.py#L398-L411

For step two, we'll need to update ApiJob so that it emits a new signal to the Controller if it's the last job of a sync here: https://github.com/freedomofpress/securedrop-client/blob/fb745f218935559d2e703a4303f429bc1750dd81/securedrop_client/api_jobs/base.py#L64 and the Controller will need to communicate this back to the GUI.

**Comment by eloquence** [comment]
Just leaving a note here that a new MetadataSync can happen while messages/replies are still being downloaded from a previous sync. If that happens, we want to make sure that the "pulse" animation that currently shows during a MetadataSync does not override the animation we want to display while messages/replies are being downloaded.

**Comment by eloquence** [comment]
Unfortunately, the full scope of this issue is likely larger than what we're able to do with the time we have left before the pilot. For the 2/20-3/4 sprint, we've therefore agreed to just reinstate a brief message that appears when we detect that there _are_ messages to be downloaded. 

I say "reinstate" because in 9dd0c7e86091ac89f08a7df45968a461ace8ab53 the old "Downloading new messages" logic was removed in preparation for the more complete solution. We can reinstate this logic for now, but let's change the string to "Retrieving new messages" (per Nina's valid comment that we want to avoid using the same verb we use to refer to submissions, in order to avoid confusion).

Question: Does the old check also cover new journalist replies, or only new source messages?

**Comment by eloquence** [comment]
Because the metadata sync itself can take a long time for a well-populated server (before any messages/replies are downloaded), we've discussed that a small useful iteration would be to extend the duration to the animation to the duration of the sync. We may take this on as a lower priority iteration during the 4/2-4/15 sprint. This can go towards this issue but should not close it.

**Comment by zenmonkeykstop** [comment]
Planned "GUI is frend" milestone broadly includes the functionality in question, we can discuss it while planning that one.
